### ISSUE TITLE
Babel breaks loadModules when using ES2015 presets (is-class always false)

### ISSUE THREAD

**Issue by ewrogers** [jeffijoe/awilix#28]
When running Babel with ES2015/stage-X presets, the `container.loadModules` method does not properly register ES6 classes due to how Babel transpiles them to ES5 (`is-class` returns false).

My workaround was to ditch ES2015 preset and go with ES2016/ES2017 and selectively choose our own plugins, as we have upgraded to Node 6.9.4. However, I can understand that some people may not have that luxury.

If this issue cannot be fixed directly (detection of transpiled classes), then it would be ideal if `loadModules` could take `asClass`/`asFunction`/`asValue` methods as hints in addition to the `Lifetime`.

Example:
```
    container.loadModules([
        ['services/**/*.js', Lifetime.SCOPED, asClass], // these are classes
        ['models/**/*.js', Lifetime.SCOPED, asValue], // these are exported as schemas
        ['routes/**/*.js', Lifetime.SINGLETON], // these are exported as factory functions
        ...
    ])
```

**Comment by jeffijoe** [comment]
That would mean you'd have to stick to using one or the other for each entry. Like

```
container.loadModules([
  ['services/*.js', { register: asClass }],
  ['repositories/*.js', { register: asFunction }],
  ['config/*.js', { register: asValue }]
])
```

What do you think?

**Comment by jeffijoe** [comment]
To improve detection, if `is-class` returns false, should we treat functions starting with an upper-case letter as classes? Worst case scenario the `new Func()` won't do any harm other than allocating an object that won't be used for anything, although this is negligible since you're not doing a million resolves every millisecond.

Would that make sense?

**Comment by ewrogers** [comment]
I think both of those make sense.

**Comment by jeffijoe** [comment]
Would you like to submit a PR for one/both of those? If not then I'll get to it once I have some time.

**Comment by jeffijoe** [comment]
Released the fix + the feature as 2.7.0. :shipit:
